x/upgrade/module.go

```diff
@@ -125,7 +127,8 @@ func (am AppModule) ExportGenesis(ctx sdk.Context) json.RawMessage {
 //
 // CONTRACT: this is registered in BeginBlocker *before* all other modules' BeginBlock functions
 func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
-	BeginBlocker(am.keeper, ctx, req)
+	skipUpgradeHeightsInt64 := am.keeper.ConvertIntArrayToInt64(viper.GetIntSlice(FlagUnsafeSkipUpgrade))
```

Comment:
can we do this in another way without relying on viper? Viper should only be used where the commands are defined

Reply:
I think the skip flags should be passed in as a parameter to `upgrade.NewKeeper` from the app constructor. So somehow they need to be passed from the CLI into the app constructor.

Reply:
100% we should avoid Viper here -- We can pass this to the keeper as @aaronc stated (in `app.go`).

Reply:
Updated it